File: AndroidCommonBaseFrameLibraryFromLorenWang/src/main/java/android/lorenwang/commonbaseframe/network/manage/AcbflwResponseGsonConverterFactory.java
package android.lorenwang.commonbaseframe.network.manage;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.annotation.Annotation;
import java.lang.reflect.Type;

import retrofit2.Converter;
import retrofit2.Retrofit;

/**
 * 功能作用：
 * 创建时间：2019-12-13 下午 22:16:25
 * 创建人：王亮（Loren wang）
 * 思路：
 * 方法：
 * 注意：
 * 修改人：
 * 修改时间：
 * 备注：
 */

public class AcbflwResponseGsonConverterFactory extends Converter.Factory {
    private Gson gson;

    private AcbflwResponseGsonConverterFactory(Gson gson) {
        this.gson = gson;
    }

    public static AcbflwResponseGsonConverterFactory create() {
        return new AcbflwResponseGsonConverterFactory(new Gson());
    }

    @Override
    public AcbflwResponseGsonConverter responseBodyConverter(Type type, Annotation[] annotations, Retrofit retrofit) {
        return new AcbflwResponseGsonConverter(gson.getAdapter(TypeToken.get(type)));
    }

    @Override
    public AcbflwRequestGsonConverter requestBodyConverter(Type type, Annotation[] parameterAnnotations, Annotation[] methodAnnotations, Retrofit retrofit) {
        return new AcbflwRequestGsonConverter(gson, gson.getAdapter(TypeToken.get(type)));
    }
}

File: AndroidCommonBaseFrameLibraryFromLorenWang/src/main/java/android/lorenwang/common_base_frame/network/manage/AcbflwResponseGsonConverterFactory.kt
package android.lorenwang.common_base_frame.network.manage

import com.google.gson.Gson
import com.google.gson.TypeAdapter
import com.google.gson.reflect.TypeToken
import okhttp3.RequestBody
import okhttp3.ResponseBody
import retrofit2.Converter
import retrofit2.Retrofit
import java.lang.reflect.Type

/**
 * 创建时间：2019-08-07 15:13
 * 创建人：王亮（Loren wang）
 * 功能作用：
 * 思路：
 * 方法：
 * 注意：
 * 修改人：
 * 修改时间：
 * 备注：
 */
class AcbflwResponseGsonConverterFactory private constructor(private val gson: Gson) : Converter.Factory() {
    override fun responseBodyConverter(type: Type, annotations: Array<Annotation>,
                                       retrofit: Retrofit): Converter<ResponseBody, *> {
        val adapter = gson.getAdapter(TypeToken.get(type))
        return AcbflwResponseGsonConverter<Any>(gson, adapter = adapter as TypeAdapter<Any>)
    }

    override fun requestBodyConverter(type: Type, parameterAnnotations: Array<Annotation>, methodAnnotations: Array<Annotation>, retrofit: Retrofit): Converter<*, RequestBody> {
        val adapter = gson.getAdapter(TypeToken.get(type))
        return AcbflwRequestGsonConverter<Any>(gson, adapter as TypeAdapter<Any>)
    }

    companion object {
        @JvmOverloads  // Guarding public API nullability.
        fun create(gson: Gson? = Gson()): AcbflwResponseGsonConverterFactory {
            if (gson == null) throw NullPointerException("gson == null")
            return AcbflwResponseGsonConverterFactory(gson)
        }
    }

}
